crates/services/txpool_v2/src/tests/tests_pending_pool.rs

```diff
@@ -0,0 +1,132 @@
+use fuel_core_services::Service;
+use fuel_core_types::{
+    fuel_tx::{
+        UniqueIdentifier,
+        UtxoId,
+    },
+    services::txpool::TransactionStatus,
+};
+use futures::StreamExt;
+
+use crate::{
+    tests::universe::TestPoolUniverse,
+    TxStatusMessage,
+};
+
+#[tokio::test]
+async fn test_tx_keep_missing_input() {
+    let mut universe = TestPoolUniverse::default();
+    universe.config.utxo_validation = true;
+    let timeout = tokio::time::Duration::from_secs(1);
+    universe.config.pending_pool_tx_ttl = timeout;
+
+    let (output, unset_input) = universe.create_output_and_input();
+    let tx1 = universe.build_script_transaction(None, Some(vec![output]), 10);
+    let tx_id1 = tx1.id(&Default::default());
+    let input = unset_input.into_input(UtxoId::new(tx_id1, 0));
+    let tx2 = universe.build_script_transaction(Some(vec![input]), None, 20);
+    let tx_id2 = tx2.id(&Default::default());
+
+    let service = universe.build_service(None, None);
+    service.start_and_await().await.unwrap();
+
+    // Given
+    let mut status_stream = service.shared.tx_update_subscribe(tx_id2).unwrap();
+    service.shared.try_insert(vec![tx2.clone()]).unwrap();
+
+    // When
+    tokio::time::timeout(timeout, status_stream.next())
+        .await
+        .unwrap_err();
+    service.shared.try_insert(vec![tx1.clone()]).unwrap();
+
+    universe
+        .waiting_txs_insertion(
+            service.shared.new_tx_notification_subscribe(),
+            vec![tx1.id(&Default::default()), tx2.id(&Default::default())],
+        )
+        .await;
+
+    // Then
+    let status = status_stream.next().await.unwrap();
+    assert!(matches!(
+        status,
+        TxStatusMessage::Status(TransactionStatus::Submitted { .. })
+    ));
+
+    service.stop_and_await().await.unwrap();
+}
+
+#[tokio::test]
+async fn test_tx_kept_expired() {
```

Comment:
What is the expired thing that is being kept?

It looks like we are asserting an error in the `then` section, but the name says nothing about that.

Reply:
I changed the name of the test to be more explicit.